src/altinn-studio-cli/Upgrade/Backend/v7Tov8/AppSettingsRewriter/AppSettingsRewriter.cs

```diff
@@ -100,10 +102,23 @@ private void RewriteRemoveHiddenDataSetting(string fileName, JsonObject settings
             warnings.Add($"RemoveHiddenDataPreview has unexpected value {removeHiddenDataPreviewValue.ToJsonString()} in {fileName}, expected a boolean");
             return;
         }
-
         appSettingsObject.Remove("RemoveHiddenDataPreview");
-        appSettingsObject.Add("RemoveHiddenData", removeHiddenDataValue);
-        appSettingsObject.Add("RequiredValidation", removeHiddenDataValue);
+        if (appSettingsObject.ContainsKey("RemoveHiddenData"))
+        {
+            warnings.Add($"RemoveHiddenData already exists in AppSettings, skipping. Tool would have set the value to: {removeHiddenDataValue} in {fileName}");
+        }
+        else
+        {
+            appSettingsObject.Add("RemoveHiddenData", removeHiddenDataValue);
+        }
 
+        if (appSettingsObject.ContainsKey("RequiredValidation"))
+        {
+            warnings.Add($"RequiredValidation already exists in AppSettings, skipping. Tool would have set the value to: {removeHiddenDataValue} in {fileName}");
+        }
+        else
+        {
+            appSettingsObject.Add("RequiredValidation", removeHiddenDataValue);
+        }
```

Comment:
Nice catch! This needs to be here as well. However, this will not fix the uncaught exception with duplicate keys, as it was caused by existing duplicate keys in appsettings which throws an error when trying to get any property from the json file. If you look at the exception thrown, it actually happens already at line 82 (or 84 here), on:

```cs
settings.TryGetPropertyValue("AppSettings", out var appSettingsNode);
```

The way I solved this in the frontend upgrade tool was by adding a try-catch block around this like here, in `LayoutSetUpgrader.cs:39`

```cs
// Read dataTypes array
JsonNode? dataTypes;
try
{
  appMetaJsonObject.TryGetPropertyValue("dataTypes", out dataTypes);
}
catch (Exception e)
{
  // Duplicate keys in the object will throw an exception here
  warnings.Add($"Unable to parse applicationmetadata.json, skipping layout sets upgrade, error: {e.Message}");
  return;
}
```